napari/plugins/io.py

```diff
@@ -106,6 +106,7 @@ def read_data_with_plugins(
             path_repr = repr(path)
         msg = f'No plugin found capable of reading {path_repr}.'
         logger.warn(msg)
+        raise ValueError(f'No plugin found capable of reading {path_repr}.')
```

Comment:
```suggestion
        raise ValueError(msg)
```

Reply:
Right?

Reply:
ahh right.  yeah, that was one of the things that made me start writing that list "follow up" bullet points and I never got back to it.  The issue here is that by using ValueError, we prevent the logging of other errors below.  I don't think it's a big deal at this point, since ultimately all anyone cares is that nothing worked.  (and the errors will still be in the plugin error list).  But it's an issue, and it's why we need a simple non-`raise` based notification mechanism ... perhaps we could use python's own [`warnings` system](https://docs.python.org/3.8/library/logging.html#logging.captureWarnings) like we've already done for `sys.exechook`.

anyway, here we should just delete the logger above, since it gets logged in the error handler